katsdpimageutils/primary_beam_correction.py

```diff
@@ -53,25 +72,18 @@ def get_position(path):
     return phase_center, image_wcs
 
 
-def radial_offset(phase_center, image_wcs):
-    """Compute radial offset of pixels from the phase centre in radians.
+def get_coordinates(image_wcs):
+    """Compute the coordinates where beam is sampled, in degrees.
 
     Parameters
     ----------
-    phase_center : astropy.coordinations.SkyCoord
-        phase center position
     image_wcs : astropy.wcs.wcs.WCS
         WCS keywords in the primary HDU
     """
-    # Get pixel coordinates
-    pixcrd = np.indices((image_wcs.array_shape[2], image_wcs.array_shape[3]))
-    row = np.ravel(pixcrd[0])
-    col = np.ravel(pixcrd[1])
-    # Convert pixel coordinates to world coordinates
-    p2w = image_wcs.pixel_to_world(col, row, 0, 0)[0]
-    # Compute a separation vector between phase centre and source positions in radians.
-    separation_rad = p2w.separation(phase_center).rad
-    return separation_rad
+    beamextent = 2
+    margin = np.linspace(-beamextent/2, beamextent/2, image_wcs.pixel_shape[0])
+    x, y = np.meshgrid(margin, margin)
+    return x, y
```

Comment:
This function assumes the input image is 2x2 degrees.

What about bigger or smaller images? Rectangular ones?

Perhaps a better way to do this is to keep `radial_offset` as it was before since it already deals with the image WCS issues then call `circbeam.I` below with `x=radial_offset(...)` and `y=0` - which is a simplifcation you can make since you are circularising the beam.

Reply:
Nice catch, I have fixed it